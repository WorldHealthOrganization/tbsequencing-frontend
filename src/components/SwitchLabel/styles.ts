import { css } from '@emotion/react';
import { font, fontSizes } from '../typography/fontSizes';
import appColors from '../../styles/colors';

export const container = css({
  display: 'flex',
  alignItems: 'center',
});

export const termsLink = css({
  fontSize: fontSizes.secondary,
  color: appColors.secondary.whoBlue,
  lineHeight: '20px',
  textDecoration: 'none',
  marginLeft: 4,
  fontFamily: font.regular,
  marginTop: '0.5px',
});
