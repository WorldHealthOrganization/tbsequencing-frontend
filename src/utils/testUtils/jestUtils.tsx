import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
