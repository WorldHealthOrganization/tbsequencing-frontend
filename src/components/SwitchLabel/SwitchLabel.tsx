/** @jsxImportSource @emotion/react */
import React, { ReactElement } from 'react';
import * as styles from './styles';
import SecondaryText from '../typography/SecondaryText';

const SwitchLabel = (): ReactElement => (
  <div css={styles.container}>
    <SecondaryText>Read and agree with service </SecondaryText>
    <a css={styles.termsLink} target="_blank" href="/terms-of-use"> Terms and Conditions</a>
  </div>
);

export default SwitchLabel;
