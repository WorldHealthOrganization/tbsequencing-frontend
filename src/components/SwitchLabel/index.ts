import SwitchLabel from './SwitchLabel';

export default SwitchLabel;
