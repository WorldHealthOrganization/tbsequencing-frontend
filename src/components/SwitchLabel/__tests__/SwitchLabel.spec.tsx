import React from 'react';
import { getTreeJSONWithRedux, render } from '../../../utils/testUtils/jestUtils';
import SwitchLabel from '../SwitchLabel';

describe('SwitchLabel component test', () => {
  test('renders SwitchLabel component without error', () => {
    render(
      <SwitchLabel />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <SwitchLabel />,
    );

    expect(tree).toMatchSnapshot();
  });
});
