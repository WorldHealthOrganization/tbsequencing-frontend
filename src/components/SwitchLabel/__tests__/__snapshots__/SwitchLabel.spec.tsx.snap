// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchLabel component test Should render correctly 1`] = `
<div
  className="css-1vjh161-SwitchLabel"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-cimhyc-MuiTypography-root-BaseText"
  >
    Read and agree with service 
  </p>
  <a
    className="css-k9b93x-SwitchLabel"
    href="/terms-of-use"
    target="_blank"
  >
     Terms and Conditions
  </a>
</div>
`;
